flink-connectors/flink-connector-kafka-0.10/src/main/java/org/apache/flink/streaming/connectors/kafka/internal/Kafka010Fetcher.java

```diff
@@ -78,6 +79,69 @@ public Kafka010Fetcher(
 				useMetrics);
 	}
 
+	private class KafkaConsumerRecordWrapper10 implements ConsumerRecordMetaInfo {
+		private static final long serialVersionUID = 2651665280744549935L;
+
+		private final ConsumerRecord<byte[], byte[]> consumerRecord;
+
+		public KafkaConsumerRecordWrapper10(ConsumerRecord<byte[], byte[]> consumerRecord) {
+			this.consumerRecord = consumerRecord;
+		}
+
+		@Override
+		public byte[] getKey() {
+			return consumerRecord.key();
+		}
+
+		@Override
+		public byte[] getMessage() {
+			return consumerRecord.value();
+		}
+
+		@Override
+		public String getTopic() {
+			return consumerRecord.topic();
+		}
+
+		@Override
+		public int getPartition() {
+			return consumerRecord.partition();
+		}
+
+		@Override
+		public long getOffset() {
+			return consumerRecord.offset();
+		}
+
+		@Override
+		public long getTimestamp() {
+			return Long.MIN_VALUE;
```

Comment:
Doesn't Kafka 0.10 support record timestamps?

Reply:
Yes, it certainly does, it should return the `consumerRecord.timestamp()`. I'll fix it.